src/main/java/org/assertj/core/api/Assumptions.java

```diff
@@ -1135,30 +1155,42 @@ public static OptionalLongAssert assumeThat(OptionalLong actual) {
     return asAssumption(assertionType, array(actualType), array(actual));
   }
 
-  @SuppressWarnings({ "unchecked" })
-  private static <ASSERTION> ASSERTION asAssumption(final Class<ASSERTION> assertionType,
+  private static <ASSERTION> ASSERTION asAssumption(Class<ASSERTION> assertionType,
                                                     Class<?>[] constructorTypes,
                                                     Object... constructorParams) {
     try {
-      Class<? extends ASSERTION> type = (Class<? extends ASSERTION>) CACHE.findOrInsert(Assumptions.class.getClassLoader(),
-                                                                                        new SimpleKey(assertionType),
-                                                                                        () -> createAssumption(assertionType));
-
+      Class<? extends ASSERTION> type = createAssumption(assertionType);
       Constructor<? extends ASSERTION> constructor = type.getConstructor(constructorTypes);
       return constructor.newInstance(constructorParams);
     } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException | InstantiationException e) {
       throw new RuntimeException(e);
     }
   }
 
+  @SuppressWarnings("unchecked")
   private static <ASSERTION> Class<? extends ASSERTION> createAssumption(Class<ASSERTION> assertionType) {
-
-    return byteBuddy.subclass(assertionType)
+    return (Class<? extends ASSERTION>) CACHE.findOrInsert(Assumptions.class.getClassLoader(),
+      new SimpleKey(assertionType),
+      () -> BYTE_BUDDY.subclass(assertionType)
                     .method(ElementMatchers.any())
-                    .intercept(MethodDelegation.to(new AssumptionMethodInterceptor()))
+                    .intercept(MethodDelegation.to(AssumptionMethodInterceptor.class))
                     .make()
-                    .load(Assumptions.class.getClassLoader())
-                    .getLoaded();
+                    .load(Assumptions.class.getClassLoader(), classLoadingStrategy(assertionType))
+                    .getLoaded());
+  }
+
+  private static ClassLoadingStrategy<ClassLoader> classLoadingStrategy(Class<?> assertClass) {
```

Comment:
looks like this can be moved in a `ClassLoadingStrategyFactory` class as the same code is used in `SoftProxies`, can you confirm there is no negative effects of sharing the same `PRIVATE_LOOKUP_IN` ?

Reply:
Yes it can. The essential part is that this lookup is done in the right module.